File: JAVA/2442. Count Number of Distinct Integers After Reverse Operations.java
class Solution {
    public int countDistinctIntegers(int[] n) {
        int i,x,r,l=n.length,k=1,q,y;
        int[] a=new int[l*2];
        for(i=0;i<l;i++)
        {
            a[i]=n[i];
        }
        q=l;
        for(i=0;i<l;i++)
        {
            r=0;
            x=n[i];
            while(x!=0)
            {
                y=x%10;
                r=(r*10)+y;
                x/=10;
            }
            a[q++]=r;
        }
        Arrays.sort(a);
        for(i=0;i<a.length-1;i++)
        {
            if(a[i]!=a[i+1])
            k++;
        }
        return k;
    }
}

File: C++/2442. Count Number of Distinct Integers After Reverse Operations.cpp
class Solution {
public:
    int countDistinctIntegers(vector<int>& n) {
        int l=n.size(),i,r,x,y,k=1;
        for(i=0;i<l;i++)
        {
            r=0;
            x=n[i];
            while(x!=0)
            {
                y=x%10;
                r=(r*10)+y;
                x/=10;
            }
            if(n[i]!=r)
            n.push_back(r);
        }
        sort(n.begin(),n.end());
        for(i=0;i<n.size()-1;i++)
        {
            if(n[i]!=n[i+1])
            k++;
        }
        return(k);
    }
};